bin/mympirun.py

```diff
@@ -31,92 +31,91 @@
 v3 refactored python 28/08/2012
 v4 cleanup 5/11/2013
 
-Expert mode:
-    export MYMPIRUN_MAIN_EXCEPTION=1 to show all exceptions
-
-TODO:
-    intel tuning code
-
-@author: Stijn De Weirdt, Jens Timmerman (HPC UGent / VSC)
+@author: Stijn De Weirdt
+@author: Jens Timmerman
+@author: Jeroen De Clerck
 """
 
-import sys
 import os
+import sys
 
-from vsc.utils import fancylogger
-from vsc.mympirun.mpi.factory import getinstance
-from vsc.mympirun.mpi.mpi import whatMPI
+import vsc.mympirun.mpi.mpi as mpim
 from vsc.mympirun.option import MympirunOption
-from vsc.mympirun.rm.sched import whatSched
-
-_logger = fancylogger.getLogger()
-
-
-class ExitException(Exception):
-    """Exception thrown when we wish to exit, but no real errors occured"""
-    pass
+from vsc.mympirun.factory import getinstance
+import vsc.mympirun.rm.sched as schedm
+from vsc.utils import fancylogger
 
 
 def get_mpi_and_sched_and_options():
-    """Parses the mpi and scheduler based on current environment and guesses the best one to use"""
-    scriptname, mpi, found_mpi = whatMPI(sys.argv[0])
-
-    ismpirun = scriptname == 'mpirun'
-
-    mo = MympirunOption(ismpirun=ismpirun)
-
-    if mo.args is None or len(mo.args) == 0:
-        mo.parser.print_shorthelp()
-        raise ExitException("Exit no args provided")
-
-    sched, found_sched = whatSched(getattr(mo.options, 'schedtype', None))
-
+    """
+    Selects mpi flavor and scheduler based on environment and arguments.
+
+    @return: A triplet containing the chosen mpi flavor, chosen scheduler and the MympirunOption class.
+    """
+
+    scriptname = os.path.basename(os.path.abspath(sys.argv[0]))
+    # if the scriptname is 'mpirun', its means that mympirun was called through the faked mpirun path
+    isfake = scriptname == 'mpirun'
+
+    # init generaloption with the various mpirun cli options
+    optionparser = MympirunOption(ismpirun=isfake)
+
+    # see if an mpi flavor was explicitly chosen as an argument
+    # if not, just use the mpirun that was called
+    # We are using sys.argv because generaloption depends on the the returned
+    # scriptname
+    setmpi = optionparser.options.setmpi if optionparser.options.setmpi else sys.argv[0]
+    optionparser.log.debug("mympirun will use %s as MPI flavor", setmpi)
```

Comment:
message makes no sense when `sys,argv[0]` is used. don't use teh if/else onliner, and add this message only in the if block